File: loan-service/src/main/java/com/gdb/loans/service/LoanService.java
package com.gdb.loans.service;

import com.gdb.loans.dto.request.LoanApplicationRequest;
import com.gdb.loans.dto.request.LoanDecisionRequest;
import com.gdb.loans.dto.response.LoanResponse;
import com.gdb.loans.dto.response.RepaymentResponse;
import java.math.BigDecimal;
import java.util.List;

public interface LoanService {
    LoanResponse applyForLoan(String loginId, LoanApplicationRequest request);
    List<LoanResponse> getLoansByUser(String loginId);
    List<LoanResponse> getAllLoans();
    LoanResponse getLoanById(Long id);
    LoanResponse approveLoan(Long id, LoanDecisionRequest request);
    LoanResponse rejectLoan(Long id, LoanDecisionRequest request);
    RepaymentResponse makeRepayment(Long loanId, BigDecimal amount);
    List<RepaymentResponse> getRepayments(Long loanId);
}

File: frontend/src/services/loanService.js
import { loanApi } from './apiConfig';

export const loanService = {
  applyForLoan: (data) => loanApi.post('/api/v1/loans/apply', data).then(r => r.data),
  getMyLoans: () => loanApi.get('/api/v1/loans/my').then(r => r.data),
  getAllLoans: () => loanApi.get('/api/v1/loans').then(r => r.data),
  getLoan: (id) => loanApi.get(`/api/v1/loans/${id}`).then(r => r.data),
  getLoansByUser: (loginId) => loanApi.get(`/api/v1/loans/user/${loginId}`).then(r => r.data),
  approveLoan: (id, data) => loanApi.put(`/api/v1/loans/${id}/approve`, data).then(r => r.data),
  rejectLoan: (id, data) => loanApi.put(`/api/v1/loans/${id}/reject`, data).then(r => r.data),
  makeRepayment: (id, amount) => loanApi.post(`/api/v1/loans/${id}/repay`, { amount }).then(r => r.data),
  getRepayments: (id) => loanApi.get(`/api/v1/loans/${id}/repayments`).then(r => r.data),
};

export default loanService;
